library/modules/EventManager.cpp

```diff
@@ -104,35 +124,47 @@ void DFHack::EventManager::unregister(EventType::EventType e, EventHandler handl
             continue;
         }
         i = handlers[e].erase(i);
+        eventLastTick.erase(handler);
         if ( e == EventType::TICK )
             removeFromTickQueue(handler);
     }
 }
 
 void DFHack::EventManager::unregisterAll(Plugin* plugin) {
-    for ( auto i = handlers[EventType::TICK].find(plugin); i != handlers[EventType::TICK].end(); i++ ) {
+    for ( auto i = handlers[EventType::TICK].find(plugin); i != handlers[EventType::TICK].end(); ++i ) {
```

Comment:
I've done this for any loops using iterators, as my understanding is `iter++` requires extra superfluous operations, a copy if I recall correctly.